src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -67,18 +157,20 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
 
             if (!_documentManager.TryGetDocument(request.TextDocument.Uri, out var documentSnapshot))
             {
-                return null;
+                return Array.Empty<VSReferenceItem>();
             }
 
+            // Cancels any previous Find All References requests which may still be active
+            ResetFarRequest();
+
             var projectionResult = await _projectionProvider.GetProjectionAsync(documentSnapshot, request.Position, cancellationToken).ConfigureAwait(false);
             if (projectionResult == null || projectionResult.LanguageKind != RazorLanguageKind.CSharp)
             {
-                return null;
+                return Array.Empty<VSReferenceItem>();
             }
 
             cancellationToken.ThrowIfCancellationRequested();
```

Comment:
What happens when the LSP platform cancels us? It'd corrupt the state right?

Reply:
The state is reset at the start of any request: https://github.com/dotnet/aspnetcore-tooling/pull/2112/files#diff-4d367dc180f9d8b2a85502e662441dc9R163-R164